spec/controllers/socure_webhook_controller_spec.rb

```diff
@@ -54,202 +38,175 @@
-          event_type: document_uploaded_webhook_body[:event][:eventType],
-          reference_id: document_uploaded_webhook_body[:event][:referenceId].to_s,
-        )
-      end
-
-      context 'when document capture session exists' do
-        let(:user) { create(:user) }
-        let(:document_capture_session) do
-          DocumentCaptureSession.create(user:).tap do |dcs|
-            dcs.socure_docv_transaction_token = '45ac9531-60ae-4bc7-805e-f7823e4e5545'
-          end
-        end
-
-        before do
-          request.headers['Authorization'] = socure_secret_key
-          allow(DocumentCaptureSession).to receive(:find_by).
-            and_return(document_capture_session)
-          allow(SocureDocvResultsJob).to receive(:perform_later)
-          allow(RateLimiter).to receive(:new).with(
-            {
-              user: user,
-              rate_limit_type: :idv_doc_auth,
-            },
-          ).and_return(rate_limiter)
-        end
+      context 'received with invalid webhook key' do
+        it 'returns unauthorized with a bad secret key' do
+          request.headers['Authorization'] = 'ABC123'
+          post :create, params: webhook_body
 
-        it 'increments rate limiter of correct user' do
-          expect(rate_limiter.attempts).to eq 0
-          post :create, params: document_uploaded_webhook_body
-          expect(rate_limiter.attempts).to eq 1
-          post :create, params: document_uploaded_webhook_body
-          expect(rate_limiter.attempts).to eq 2
+          expect(response).to have_http_status(:unauthorized)
         end
 
-        it 'enqueues a SocureDocvResultsJob' do
+        it 'returns unauthorized with no secret key' do
           post :create, params: webhook_body
 
-          expect(SocureDocvResultsJob).to have_received(:perform_later).
-            with(document_capture_session_uuid: document_capture_session.uuid)
+          expect(response).to have_http_status(:unauthorized)
         end
       end
 
-      context 'when document capture session does not exist' do
+      context 'with a valid webhook key' do
         before do
-          allow(NewRelic::Agent).to receive(:notice_error)
+          request.headers['Authorization'] = socure_secret_key
         end
-
-        it 'logs an error with NewRelic' do
-          request.headers['Authorization'] = socure_secret_key_queue.last
+        it 'returns OK and logs an event with a correct secret key and body' do
           post :create, params: webhook_body
 
-          expect(NewRelic::Agent).to have_received(:notice_error)
+          expect(response).to have_http_status(:ok)
+          expect(@analytics).to have_logged_event(
+            :idv_doc_auth_socure_webhook_received,
+            created_at: '2020-01-01T00:00:00Z',
+            customer_user_id:,
+            docv_transaction_token: event_docv_transaction_token,
+            event_type:,
+            reference_id: reference_id,
```

Comment:
If we're going to use the compact form, let's use it whenever we can.

```suggestion
            event_type:,
            reference_id:,
```